examples/connect.rs

```diff
@@ -78,60 +97,111 @@ impl fmt::Display for PeerId {
 }
 
 fn main() {
-    let mut event_loop = unwrap!(Core::new());
-    let service_id = rand::thread_rng().gen::<PeerId>();
-    println!("Service id: {}", service_id);
-
-    let config =
-        unwrap!(
-        ConfigFile::open_path(PathBuf::from("sample.config")),
-        "Failed to read crust config file: sample.config",
-    );
-    let make_service = Service::with_config(&event_loop.handle(), config, service_id);
-    let service =
-        unwrap!(
-        event_loop.run(make_service),
-        "Failed to create Service object",
-    );
-
-    let listeners =
-        unwrap!(
-        event_loop.run(service.start_listening().collect()),
-        "Failed to start listening to peers",
-    );
-    for listener in &listeners {
-        println!("Listening on {}", listener.addr());
+    let args: Args = {
+        Docopt::new(USAGE)
+        .and_then(|d| d.deserialize())
+        .unwrap_or_else(|e| e.exit())
+    };
+
+    let config = unwrap!(ConfigFile::new_temporary());
+    if let Some(rendezvous_addr) = args.flag_rendezvous_peer {
+        unwrap!(config.write()).hard_coded_contacts = vec![rendezvous_addr];
+    }
+    if args.flag_disable_tcp {
+        unwrap!(config.write()).dev = Some(DevConfigSettings {
+            disable_tcp: true,
+            .. Default::default()
+        });
     }
 
-    let our_conn_info =
-        unwrap!(
-        event_loop.run(service.prepare_connection_info()),
-        "Failed to prepare connection info",
-    );
-    let pub_conn_info = our_conn_info.to_pub_connection_info();
-    println!(
-        "Public connection information:\n{}\n",
-        unwrap!(serde_json::to_string(&pub_conn_info))
-    );
-
-    println!("Enter remote peer public connection info:");
-    let their_info = readln();
-    let their_info: PubConnectionInfo<PeerId> = unwrap!(serde_json::from_str(&their_info));
-
-    let peer =
-        unwrap!(
-        event_loop.run(service.connect(our_conn_info, their_info)),
-        "Failed to connect to given peer",
-    );
-    println!("Connected to peer: {}", peer.uid());
-
-    // Run event loop forever.
-    let res = event_loop.run(empty::<(), ()>());
-    unwrap!(res);
+    let mut core = unwrap!(Core::new());
```

Comment:
I intentionally wanted to keep the very first example (`connect`) simple - friendlier for newcomers.
Could we have to separate examples for this?